File: ios/RNHelloDoctor/Video/HDVideoLocalViewManager.m
//
//  HDOLocalVideoViewManager.m
//  hellodoctor
//
//  Created by HelloDoctor on 9/12/20.
//

#import <Foundation/Foundation.h>
#import <CoreMedia/CMFormatDescription.h>
#import <React/RCTViewManager.h>
#import <TwilioVideo/TwilioVideo.h>
#import "HDVideoLocalViewManager.h"
#import "RNHelloDoctor-Swift.h"

@interface HDVideoLocalViewManager()

@end

@implementation HDVideoLocalViewManager

//+ (BOOL)requiresMainQueueSetup
//{
//    return YES;  // only do this if your module initialization relies on calling UIKit!
//}

RCT_EXPORT_MODULE()

- (UIView *)view
{
    NSLog(@"[HDVideoLocalViewManager] creating new view");

    UIView *container = [[UIView alloc] init];
    TVIVideoView *inner = [[TVIVideoView alloc] init];
    inner.autoresizingMask = (UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth);
    inner.contentMode = UIViewContentModeScaleAspectFill;

    [container addSubview:inner];

    HDVideo* hdVideo = [HDVideo getInstance];
    [hdVideo setLocalViewWithView:inner];

    return container;
}

@end

File: ios/RNHelloDoctor/Video/HDVideoLocalViewManager.swift
//
//  HDVideoLocalViewManager.swift
//  RNHelloDoctor
//
//  Created by HelloDoctor on 3/29/22.
//

import Foundation
import TwilioVideo
import UIKit

@objc(HDVideoLocalViewManager)
class HDVideoLocalViewManager: RCTViewManager {
    @objc
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func view() -> UIView! {
        let container = UIView()
        let inner = VideoView()
        inner.autoresizingMask = [.flexibleHeight, .flexibleWidth]
        inner.contentMode = .scaleAspectFill
        
        container.addSubview(inner)
        
        let hdVideo = HDVideo.getInstance()
        hdVideo.setLocalView(view:inner)
        
        return container
    }
}
